store/src/java/com/zimbra/cs/service/mail/ToXML.java

```diff
@@ -1631,7 +1621,36 @@ private static Element encodeMessageAsMPHelper(boolean bestEffort, Element paren
             List<MPartInfo> parts = Mime.getParts(mm, getDefaultCharset(msg));
             if (parts != null && !parts.isEmpty()) {
                 Set<MPartInfo> bodies = Mime.getBody(parts, wantHTML);
-                addParts(m, parts.get(0), bodies, part, maxSize, neuter, false, getDefaultCharset(msg), bestEffort, wantContent);
+                // added condition to check if email is external and if it contains warning in html content
+                if (parts.size() > 1) {
+                    if (ExternalEmailWarning.getInstance().isEnabled()) {
+                        List<String> emails =  getSenderReceiverEmailAddresses(m);
+                        if (emails.size() >= 2) {
+                            ZimbraLog.lmtp.info(" originator email : "+emails.get(0));   //originator's email
+                            ZimbraLog.lmtp.info(" receiver email : "+emails.get(1));   //receivers's email
+                            boolean isExternal = ExternalEmailWarning.getInstance().isExternal(emails.get(1),emails.get(0));
+                            if (isExternal) {
+                                String textContent = parts.get(1).getMimePart().getContent().toString();
+                                String htmlContent = parts.get(2).getMimePart().getContent().toString();
+                                String[] textContents = textContent.split("\n", 2);
+                                String baseEmailWarning = textContents[0];
+                                ZimbraLog.lmtp.info("Base Email Warning : "+baseEmailWarning);
+                                if (!htmlContent.contains(baseEmailWarning)) {
```

Comment:
check below scenario and how it will behave with this code
1) admin sets warning for external mails, let's say as "Disclaimer 1"
2) user sends some mails
3) admin updates warning for external mails, let's say as "Disclaimer 2"
4) if user opens mails sent in step 2, what he will see Disclaimer 1 or 2?

Reply:
User will see "Disclaimer 1"
During the api call to fetch the message, message are extracted from storage. Since Disclaimer 1 one was the warning when email was saved, it will fetched and passed back in response.